app/src/modules/settings/routes/data-model/field-detail/shared/related-field-select.vue

```diff
@@ -94,7 +94,7 @@ export default defineComponent({
 					!!field.schema?.is_primary_key ||
 					props.disabledFields.includes(field.field) ||
 					props.typeDenyList.includes(field.type) ||
-					!props.typeAllowList.includes(field.type),
+					(props.typeAllowList.length ? !props.typeAllowList.includes(field.type) : false),
```

Comment:
`typeAllowList` should default to an empty array when it's not provided in props. When is it not an array?

Reply:
I agree that `typeAllowList` should default to an empty array when it's not provided in props. To solve this issue, I think we have to pass `typeAllowList` prop with values. Let me know about it. I have attached a screenshot for `type-allow-list` prop passed in various routes. 

![image](https://user-images.githubusercontent.com/93135924/213438940-9377864a-d4dc-4e09-9d25-c168633211c4.png)